### PR TITLE
[MM-31929] server: add mlog.Error review feautre

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Mattermod will remind us to review our logging guidelines.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31929

### CODE REVIEW

File: server/review.go
Comment (isacikgoz) [inline server/review.go:20]: I will update this after the https://mattermost.atlassian.net/browse/MM-32010 Also I'm open to better sentences 😅

Comment (agnivade) [reply server/review.go:20]: Yeah, we should add the contents of the document to this section: https://github.com/mattermost/mattermost-developer-documentation/pull/751. Shall we wait until the PR is merged?

Comment (isacikgoz) [reply server/review.go:20]: Deal 🤝

Comment (agnivade) [reply server/review.go:20]: Actually, that PR will take some time. We can go ahead with this pointing to the google doc for now if you prefer.

File: server/review.go
Comment (agnivade) [inline server/review.go:77]: We should definitely pass the context from the upper layer in this HTTP call. We don't want to keep it stuck forever.

File: server/review.go
Comment (agnivade) [inline server/review.go:67]: Does this mean that we will be only commenting once even if the PR contains multiple lines of `mlog.Error/Critical`?

File: server/review.go
Comment (agnivade) [inline server/review.go:43]: line == ""

File: server/review.go
Comment (agnivade) [inline server/review.go:30]: Maybe worth adding some error context here.

File: server/review.go
Comment (isacikgoz) [reply server/review.go:67]: Yes, I think it will be enough to point out the guideline.

File: server/review.go
Comment (isacikgoz) [reply server/review.go:43]: 🤦

**Review by agnivade — APPROVED** [review]

**Review by nronas — APPROVED** [review]
LGTM 🚢 - it

**Comment by agnivade** [comment]
@isacikgoz - What is this waiting for?

**Comment by isacikgoz** [comment]
@agnivade This is waiting for the docs PR to be merged. 😬

**Pull request merged**
